### PR TITLE
protect lunatik_runtime() call

### PR DESCRIPTION
**Pull request by lneto**
* change ENINVAL to ENOEXEC
* replace lunatik.runtimes() with lunatik.runtimes
* require libs automatically on lunatik_pushobject()

### CODE REVIEW

File: lunatik_core.c [hunk 1]
```diff
@@ -194,21 +196,15 @@ static int lunatik_newruntime(lunatik_object_t **pruntime, lua_State *parent, co
...
-	luaL_openlibs(L);
-	if (sleep) {
-		luaL_requiref(L, "lunatik", luaopen_lunatik, 0);
-		lua_pop(L, 1); /* lunatik library */
-	}
-
-	if (lunatik_dofile(L, script) != LUA_OK) {
-		lunatik_runerror(L, parent, lua_tostring(L, -1));
+	lua_pushcfunction(L, lunatik_runscript);
+	lua_pushlightuserdata(L, (void *)script);
```
Comment (sav) [inline lunatik_core.c:200]: not a factual suggestion for this review, per se, but rather, just to mention: about all these type casts, I really like those `cast_*` macros Lua define in `lua/llimits.h`; maybe Lunatik could have their own.

**Review by sav — APPROVED** [review]

File: lunatik.h [hunk 2]
```diff
@@ -180,10 +180,18 @@ int lunatik_monitorobject(lua_State *L);
 #define lunatik_getobject(o)		kref_get(&(o)->kref)
 #define lunatik_putobject(o)		kref_put(&(o)->kref, lunatik_releaseobject)
 
+static inline void lunatik_require(lua_State *L, const char *libname)
```
[the remark this answers is not in the document]
Comment (lneto) [reply lunatik.h:183]: problem here is that it's now used by `pushobjects`.. will need to move both around "dirtying" the patch.. if you think it's worth, I can move..

File: lunatik_core.c (hunk 1 above)
Comment (lneto) [reply lunatik_core.c:200]: we could use it if you think it's worth.. I wouldn't replicate them though.. we could include this header on lunatik.h perhaps..

Comment (lneto) [reply lunatik_core.c:200]: but I would apply this to the code as a whole, not only in this single place now.. what do you think?

File: lunatik.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (sav) [reply lunatik.h:183]: no worries.

File: lunatik_core.c (hunk 1 above)
Comment (sav) [reply lunatik_core.c:197]: I just mentioned as something to consider for future changes.

Comment (lneto) [reply lunatik_core.c:197]: btw, you can create a issue for that..

**Pull request merged**